src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/quota/AzureQuotaPlugin.java

```diff
@@ -0,0 +1,296 @@
+    ResourceAllocation buildQuota(ComputeAllocation computeAllocation,
+                                  NetworkAllocation networkAllocation,
+                                  PublicIpAllocation publicIpAllocation,
+                                  VolumeAllocation volumeAllocation) {
+        int instances = computeAllocation.getInstances();
+        int ram = computeAllocation.getRam();
+        int vCPU = computeAllocation.getvCPU();
+        int volumes = volumeAllocation.getInstances();
+        int storage = volumeAllocation.getStorage();
+        int networks = networkAllocation.getInstances();
+        int publicIps = publicIpAllocation.getInstances();
+
+        return ResourceAllocation.builder()
+                .instances(instances)
+                .vCPU(vCPU)
+                .ram(ram)
+                .volumes(volumes)
+                .storage(storage)
+                .networks(networks)
+                .publicIps(publicIps)
+                .build();
+    }
+
+    @VisibleForTesting
+    Map<String, ComputeUsage> getComputeUsageMap(Azure azure) {
+        Map<String, ComputeUsage> computeUsageMap = new HashMap<>();
+        List<String> validComputeUsages = Arrays.asList(QUOTA_VM_INSTANCES_KEY, QUOTA_VM_CORES_KEY);
+
+        this.getComputeUsage(azure)
+                .stream()
+                .filter(computeUsage -> validComputeUsages.contains(computeUsage.name().value()))
+                .forEach(computeUsage -> computeUsageMap.put(computeUsage.name().value(), computeUsage));
+
+        return computeUsageMap;
+    }
+
+    @VisibleForTesting
+    PagedList<ComputeUsage> getComputeUsage(Azure azure) {
+        return azure.computeUsages().listByRegion(this.defaultRegionName);
+    }
+
+    @VisibleForTesting
+    Map<String, NetworkUsage> getNetworkUsageMap(Azure azure) {
+        Map<String, NetworkUsage> networkUsageMap = new HashMap<>();
+        List<String> validNetworkUsages = Arrays.asList(QUOTA_NETWORK_INSTANCES, QUOTA_PUBLIC_IP_ADDRESSES);
+
+        this.getNetworkUsage(azure)
+                .stream()
+                .filter(networkUsage -> validNetworkUsages.contains(networkUsage.name().value()))
+                .forEach(networkUsage -> networkUsageMap.put(networkUsage.name().value(), networkUsage));
+
+        return networkUsageMap;
+    }
+
+    @VisibleForTesting
+    PagedList<NetworkUsage> getNetworkUsage(Azure azure) {
+        return azure.networkUsages().listByRegion(this.defaultRegionName);
+    }
+
+    @VisibleForTesting
+    int getMemoryUsage(Azure azure) {
+        List<String> sizeNamesInUse = this.getVirtualMachineSizeNamesInUse(azure);
+        Map<String, VirtualMachineSize> virtualMachineSizes = this.getVirtualMachineSizes(sizeNamesInUse, azure);
+        return this.doGetMemoryUsage(sizeNamesInUse, virtualMachineSizes);
+    }
+
+    /**
+     * Return the total memory in use by all created virtual machines
+     *
+     * @param sizeNamesInUse list of virtual machine size names
```

Comment:
Are we using this pattern in the other parts of the Fogbow code? I mean in the commentary, the usage of the @param and @return